### DISCUSSION TITLE
Would y'all want to collaborate on MST docs?

### DISCUSSION THREAD

**Discussion by coolsoftwaretyler** [ConrabOpto/mst-query#63] [Q&A] [answered]
Hey folks,

The MobX-State-Tree maintainers want to improve our docs in 2024.

Right now we have just one link to your library in the MobX-State-Tree docs. I think there's room for a lot more.

Since so many people come to us asking for help with things that MST query is designed to help with, I'd love to get some examples, articles, tutorials, etc from y'all that we can showcase.

Is that interesting to you? Is there some other way you'd prefer to collaborate (if at all?)

Let me know! My email is on my GitHub profile if anyone wants to connect one-on-one.

Thanks!

**Comment by k-ode** [comment]
Hi @coolsoftwaretyler !

We'd definitely be interested in helping out with improving the docs for Mobx-State-Tree.

To me it would make a lot sense to showcase how mst-query can help with simplyifing async flows. In such a section, you could first explain the normal mst patterns, and then have another part that shows corresponding code but with mst-query.

I also think it would be great to somehow show how mst-query can help you normalize nested data like you get from a graphql endpoint. Not sure if this would belong in a "Working with graphql"- section or "Normalizing data"-section, but hopefully you get the idea.

One thing to note is that mst-query right now is tightly tied to React. This might be an issue unless you highlight mst-query in a section about React integration. It should technically be possible to separate the parts that depend on React from the integration into mst, but at this moment I'm not sure how much work that would be.

**Reply by coolsoftwaretyler** [reply]
Hey @k-ode - thanks so much!

I'm not so much worried about the React dependency. I'd love to feature y'all in some kind of "recipe" section, where we do exactly as you outlined: explain a common use case (work with graphql or normalize data), show how one might do it with MST in a bespoke way, and then show the better mst-query way.

As a recipe, and less of MobX-State-Tree "doc", I think it's reasonable to just say "mst-query depends on React for now". Folks can take it or leave it, and I think there's enough React out there to make a recipe like this extremely helpful.

Would you be interested in outlining, drafting, or writing this (or some subset of the three?) If so, I'd love a PR against MST with just a markdown file we can edit and iterate on - whatever you have interest and capacity for.

**Reply by k-ode** [reply]
@coolsoftwaretyler Great, I'll try and get a PR going shortly.

**Comment by k-ode** [comment, marked as answer]
The documentation is here: https://mobx-state-tree.js.org/recipes/mst-query
